### PR TITLE
Auto refresh for main screen

### PR DESCRIPTION
**Pull request by giladgo**

### CODE REVIEW

File: app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java [hunk 1]
```diff
@@ -50,34 +54,30 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-        super.onResume();
-        refreshCurrentMatch();
+    public void onStop() {
+        super.onStop();
+        mEventBus.unregister(this);
     }
 
+    @Subscribe
+    public void refreshCurrentMatch(CurrentSessionStore.SessionUpdateEvent event) {
+        mCurrentMatch = ((Application)getActivity().getApplication()).getCurrentSessionStore().getCurrentMatch();
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java:76]: maybe extract the session store to a field and set it onstart?

File: app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java [hunk 2]
```diff
@@ -50,34 +54,30 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View rootView = inflater.inflate(R.layout.fragment_current_match, container, false);
         ButterKnife.inject(this, rootView);
+        mEventBus = ((Application)getActivity().getApplication()).getCurrentSessionStore().getEventBus();
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java:57]: this is a pretty nasty line and you use it multiple times.
how about 
`SessionStore.getEvenBus(getActivity)`

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 3]
```diff
@@ -77,6 +73,25 @@ public Void then(Task<Void> task) throws Exception {
         return rootView;
     }
 
+
+    @Subscribe
+    public void refreshQueue(CurrentSessionStore.SessionUpdateEvent event) {
+        mQueueItems = ((Application)getActivity().getApplication()).getCurrentSessionStore().getQueue();
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:79]: `SessionStore.current(getActivity().getQueue()`

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 4]
```diff
@@ -95,117 +110,72 @@ public void run() {
...
-            return mNonDismissedPositions.get(position);
+            return mQueueItems.get(position);
         }
 
-        public void dismissItem(int position) {
-            mNonDismissedPositions.remove(position);
+        @Override
+        public long getItemId(int position) {
+            return 0;
         }
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:161]: needed?

File: app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            @Override
+            public Void then(Task<Pair<Match, List<QueueItem>>> pairTask) throws Exception {
+                Pair<Match, List<QueueItem>> pair = pairTask.getResult();
+                mCurrentMatch = pair.first;
+                mQueue = pair.second;
+
+                mEventBus.post(produceSessionUpdateEvent());
+                return null;
+            }
+        }, Task.UI_THREAD_EXECUTOR);
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java:59]: why on UI thread?

File: app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    public Bus getEventBus() {
+        return mEventBus;
+    }
+
+
+    public static class SessionUpdateEvent {
+    }
+
+    public CurrentSessionStore(Bus eventBus, Bus broadcastEventBus) {
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java:39]:
```
CurrentSessionStore(Bus broadcastBus)
```

and create the internal bus, internally

File: app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java (hunk 5 above)
Comment (udiWertheimer) [inline app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java:59]: Why UI thread?

Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java:59]: The event bus works on the UI thread.

**Comment by yonbergman** [comment]
sbab

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java (hunk 4 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:161]: Yep, it's abstract in the base class.

File: app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java (hunk 2 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java:57]: Will that go to the application and get the instance?

**Comment by udiWertheimer** [comment]
grrrrreat

File: app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java (hunk 1 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java:76]: :+1:

File: app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java (hunk 6 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/data/CurrentSessionStore.java:39]: :+1:

File: app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java (hunk 2 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/CurrentMatchFragment.java:57]: :+1:

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java (hunk 3 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:79]: :+1:

**Pull request merged**